### PR TITLE
[frontent/backend] Enforcing schema to generate mappings and prevent any type confusion

### LINKED ISSUE #5377: Can't create activity digest
## Reproducible Steps
- Go to Settings > Activity (EE should be enabled)
- Go to Alerting
- Create an activity trigger
- Create an activity digest
- An error occurs
image

### LINKED ISSUE #5384: Can't update log sources of a course of action
Go to Techniques > Courses of action
Edit a course of action.
Edit the 'log sources' field to add sources.
You can't edit the field: 
image

### LINKED ISSUE #5386: Can't create a threat actor individual with roles
Create a threat actor individual. In the creation form, enter a name and values for 'roles' (in the 'details' tab).
Error occurs when you click on the 'create' button : 
image

### LINKED ISSUE #5392: Enforcing schema to generate mappings and prevent any type confusion
## Use case

Enforcing schema to generate mappings and prevent any type confusion.
Related to https://github.com/OpenCTI-Platform/opencti/pull/5340

### PR DESCRIPTION
**Pull request by Archidoit**
Schema refacto
review by Julien, Cathia & Laurent

### CODE REVIEW

File: opencti-platform/opencti-front/src/private/components/common/form/HeightField.tsx [hunk 1]
```diff
@@ -85,7 +85,7 @@ export const HeightFieldEdit: FunctionComponent<HeightFieldEditProps> = ({
                             input: {
                               key: 'height',
                               value: [heightToPivotFormat(measure)],
-                              object_path: `[${height.index}].measure`,
+                              object_path: `/height/${height.index}/.measure`,
```
Comment (labo-flg) [inline opencti-platform/opencti-front/src/private/components/common/form/HeightField.tsx]:
```suggestion
                              object_path: `/height/${height.index}/measure`,
```

File: opencti-platform/opencti-graphql/src/modules/attributes/stixSightingRelationship-registrationAttributes.ts [hunk 2]
```diff
@@ -5,12 +5,12 @@ import { STIX_SIGHTING_RELATIONSHIP } from '../../schema/stixSightingRelationshi
...
-    { name: 'x_opencti_negative', type: 'boolean', mandatoryType: 'customizable', editDefault: true, multiple: false, upsert: false, label: 'False positive' },
-    { name: 'x_opencti_workflow_id', type: 'string', mandatoryType: 'no', editDefault: false, multiple: false, upsert: false },
+    { name: 'attribute_count', label: 'Count', type: 'numeric', precision: 'integer', mandatoryType: 'external', editDefault: true, multiple: false, upsert: false, isFilterable: true },
+    { name: 'first_seen', label: 'First seen', type: 'date', mandatoryType: 'customizable', editDefault: true, multiple: false, upsert: false, isFilterable: true },
+    { name: 'last_seen', label: 'Last seen', type: 'date', mandatoryType: 'customizable', editDefault: true, multiple: false, upsert: false, isFilterable: true },
+    { name: 'description', label: 'Description', type: 'string', format: 'short', mandatoryType: 'customizable', editDefault: true, multiple: false, upsert: true, isFilterable: true },
+    { name: 'x_opencti_negative', label: 'Negative', type: 'boolean', mandatoryType: 'customizable', editDefault: true, multiple: false, upsert: false, isFilterable: true },
```
Comment (labo-flg) [inline opencti-platform/opencti-graphql/src/modules/attributes/stixSightingRelationship-registrationAttributes.ts]: maybe label should be "False Positive" to match before?

Comment (Archidoit) [reply opencti-platform/opencti-graphql/src/modules/attributes/stixSightingRelationship-registrationAttributes.ts]: yes maybe, I'm going to modify the label

Comment (Archidoit) [reply opencti-platform/opencti-graphql/src/modules/attributes/stixSightingRelationship-registrationAttributes.ts]: fixed

**Comment by Archidoit** [comment]
Can' create a user (bulk indexing fail).

File: opencti-platform/opencti-graphql/src/migrations/1697374242887-bio_mappings.js [hunk 3]
```diff
@@ -1,25 +0,0 @@
-import { elUpdateIndicesMappings } from '../database/engine';
-
-export const up = async (next) => {
-  await elUpdateIndicesMappings({
```
Comment (SouadHadjiat) [inline opencti-platform/opencti-graphql/src/migrations/1697374242887-bio_mappings.js:4]: why do we remove these migrations ? What would happen for old platforms when you upgrade ?

**Comment by Archidoit** [comment]
Can't create a case template (in Taxonomie) : 
image

File: opencti-platform/opencti-graphql/src/migrations/1697374242887-bio_mappings.js (hunk 3 above)
Comment (richard-julien) [reply opencti-platform/opencti-graphql/src/migrations/1697374242887-bio_mappings.js:4]: At start the platform check the index and complete them with the new attributes define in the model automatically

Comment (SouadHadjiat) [reply opencti-platform/opencti-graphql/src/migrations/1697374242887-bio_mappings.js:4]: oh I guess the refresh mapping will do the work at platform start

**Comment by Archidoit** [comment]
Can't create a workbench: 
image

**Comment by Archidoit** [comment]
Can't create an entity with participants. (ex: try to create an incident with participant=admin)
image

**Comment by Archidoit** [comment]
Can't create an incident response: 
image

Same for Case RFI and Case RFT

**Comment by labo-flg** [comment]
Data import (CSV mapper, large test sample) throws an error in the worker

  `{'name': 'DATABASE_ERROR', 'message': 'Update indexing fail'}` 
  
  in the end (import was still successfull)

```
{"timestamp": "2024-01-04T11:00:08.522474Z", "level": "INFO", "name": "api", "message": "Report expectation", "attributes": {"work_id": "work_d336676c-4ee5-4257-96ff-b2a86688d4af_2024-01-04T10:59:06.025Z"}}
{"timestamp": "2024-01-04T11:00:08.582871Z", "level": "ERROR", "name": "api", "message": "Cannot report expectation", "exc_info": "Traceback (most recent call last):\n  File \"C:\\Users\\LaurentBonnet\\scoop\\apps\\python310\\current\\lib\\site-packages\\pycti\\api\\opencti_api_work.py\", line 56, in report_expectation\n    self.api.query(query, {\"id\": work_id, \"error\": error})\n  File \"C:\\Users\\LaurentBonnet\\scoop\\apps\\python310\\current\\lib\\site-packages\\pycti\\api\\opencti_api_client.py\", line 344, in query\n    raise ValueError(\nValueError: {'name': 'DATABASE_ERROR', 'message': 'Update indexing fail'}"}
{"timestamp": "2024-01-04T11:00:08.600388Z", "level": "INFO", "name": "worker", "message": "Message processed, thread terminated"}
```

Log in the octi backend:

```
{"category":"APP","errors":[{"attributes":{"body":{"script":{"lang":"painless","params":{"completed_number":"1","now":"2024-01-04T11:00:08.527Z"},"source":"ctx._source['status'] = \"complete\";\n      ctx._source['completed_number'] = params.completed_number;\n      ctx._source['completed_time'] = params.now;"}},"documentId":"work_d336676c-4ee5-4257-96ff-b2a86688d4af_2024-01-04T10:59:06.025Z","genre":"TECHNICAL","http_status":500},"message":"Update indexing fail","name":"DATABASE_ERROR","stack":"DATABASE_ERROR: Update indexing fail\n    at error (C:\\Users\\LaurentBonnet\\IdeaProjects\\opencti\\opencti-platform\\opencti-graphql\\src\\config\\errors.js:8:10)\n    at DatabaseError (C:\\Users\\LaurentBonnet\\IdeaProjects\\opencti\\opencti-platform\\opencti-graphql\\src\\config\\errors.js:58:48)\n    at C:\\Users\\LaurentBonnet\\IdeaProjects\\opencti\\opencti-platform\\opencti-graphql\\src\\database\\engine.js:2461:11\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at reportExpectation (C:\\Users\\LaurentBonnet\\IdeaProjects\\opencti\\opencti-platform\\opencti-graphql\\src\\domain\\work.js:220:5)"},{"message":"document_parsing_exception\n\tCaused by:\n\t\tillegal_argument_exception: Integer value passed as String\n\tRoot causes:\n\t\tdocument_parsing_exception: [1:599] failed to parse field [completed_number] of type [integer] in document with id 'work_d336676c-4ee5-4257-96ff-b2a86688d4af_2024-01-04T10:59:06.025Z'. Preview of field's value: '1'","name":"ResponseError","stack":"ResponseError: document_parsing_exception\n\tCaused by:\n\t\tillegal_argument_exception: Integer value passed as String\n\tRoot causes:\n\t\tdocument_parsing_exception: [1:599] failed to parse field [completed_number] of type [integer] in document with id 'work_d336676c-4ee5-4257-96ff-b2a86688d4af_2024-01-04T10:59:06.025Z'. Preview of field's value: '1'\n    at _SniffingTransport.request (C:\\Users\\LaurentBonnet\\IdeaProjects\\opencti\\opencti-platform\\opencti-graphql\\node_modules\\@elastic\\transport\\src\\Transport.ts:553:17)\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at _Client.UpdateApi [as update] (C:\\Users\\LaurentBonnet\\IdeaProjects\\opencti\\opencti-platform\\opencti-graphql\\node_modules\\@elastic\\elasticsearch\\src\\api\\api\\update.ts:75:10)\n    at reportExpectation (C:\\Users\\LaurentBonnet\\IdeaProjects\\opencti\\opencti-platform\\opencti-graphql\\src\\domain\\work.js:220:5)"}],"inner_relation_creation":0,"level":"warn","message":"Update indexing fail","operation":"reportExpectation","operation_query":"mutation reportExpectation($id:ID!$error:WorkErrorInput){workEdit(id:$id){reportExpectation(error:$error)}}","size":88,"time":53,"timestamp":"2024-01-04T11:00:08.578Z","type":"WRITE_ERROR","user":{"applicant_id":"88ec0c6a-13ce-5e39-b486-354fe4a7084f","call_retry_number":"1","group_ids":["dbee0af4-7a7c-4722-ae3f-7e65202ccd2f"],"ip":"::1","organization_ids":[],"socket":"query","user_id":"88ec0c6a-13ce-5e39-b486-354fe4a7084f"},"version":"5.12.15"}
```

**Comment by labo-flg** [comment]
> Can' create a user (bulk indexing fail).

Fixed with latest changes

> Can't create a case template (in Taxonomie) :

Fixed with latest changes

> Can't create a workbench:

Fixed with latest changes

> Can't create an entity with participants. (ex: try to create an incident with participant=admin)

Fixed with latest changes

> Can't create an incident response:

Fixed with latest changes

> Same for Case RFI and Case RFT

Fixed with latest changes

> Data import (CSV mapper, large test sample) throws an error in the worker

Fixed with latest changes

**Comment by labo-flg** [comment]
Adding a Threat actor group to a container  (a previously created incident response) - including neighbours, fails with `Simple indexing failed` 

error details : `mapping set to strict, dynamic introduction of [options] within [actions.context] is not allowed"`

backend logs:

```
{"category":"APP","errors":[{"attributes":{"body":{"actions":[{"context":{"field":"container-object","options":{"includeNeighbours":true},"type":"ATTRIBUTE","values":["6df9caba-68b9-4840-abfc-2653e7fb8419"]},"type":"ADD"}],"authorized_authorities":["KNOWLEDGE_KNUPDATE"],"authorized_members":[{"access_right":"admin","id":"88ec0c6a-13ce-5e39-b486-35
4fe4a7084f"}],"completed":false,"created_at":"2024-01-05T10:03:02.572Z","entity_type":"BackgroundTask","errors":[],"id":"77807009-0731-4bef-a8d6-d409b3d8134e","initiator_id":"88ec0c6a-13ce-5e39-b486-354fe4a7084f","internal_id":"77807009-0731-4bef-a8d6-d409b3d8134e","last_execution_date":null,"scope":"KNOWLEDGE","standard_id":"backgroundtask--3d20d
667-f291-5311-a26d-817e5235346c","task_expected_number":1,"task_ids":["9c393121-5c90-4049-9944-095185ea7b6e"],"task_position":null,"task_processed_number":0,"type":"LIST"},"genre":"TECHNICAL","http_status":500},"message":"Simple indexing fail","name":"DATABASE_ERROR","stack":"DATABASE_ERROR: Simple indexing fail\n    at error (C:\\Users\\LaurentBo
nnet\\IdeaProjects\\opencti\\opencti-platform\\opencti-graphql\\src\\config\\errors.js:8:10)\n    at DatabaseError (C:\\Users\\LaurentBonnet\\IdeaProjects\\opencti\\opencti-platform\\opencti-graphql\\src\\config\\errors.js:58:48)\n    at C:\\Users\\LaurentBonnet\\IdeaProjects\\opencti\\opencti-platform\\opencti-graphql\\src\\database\\engine.js:24
99:11\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at elIndex (C:\\Users\\LaurentBonnet\\IdeaProjects\\opencti\\opencti-platform\\opencti-graphql\\src\\database\\engine.js:2498:3)\n    at createListTask (C:\\Users\\LaurentBonnet\\IdeaProjects\\opencti\\opencti-platform\\opencti-graphql\\src\\domain\\backgroundTas
k-common.js:184:3)"},{"message":"strict_dynamic_mapping_exception\n\tRoot causes:\n\t\tstrict_dynamic_mapping_exception: [1:738] mapping set to strict, dynamic introduction of [options] within [actions.context] is not allowed","name":"ResponseError","stack":"ResponseError: strict_dynamic_mapping_exception\n\tRoot causes:\n\t\tstrict_dynamic_mappin
g_exception: [1:738] mapping set to strict, dynamic introduction of [options] within [actions.context] is not allowed\n    at _SniffingTransport.request (C:\\Users\\LaurentBonnet\\IdeaProjects\\opencti\\opencti-platform\\opencti-graphql\\node_modules\\@elastic\\transport\\src\\Transport.ts:553:17)\n    at processTicksAndRejections (node:internal/p
rocess/task_queues:95:5)\n    at _Client.IndexApi [as index] (C:\\Users\\LaurentBonnet\\IdeaProjects\\opencti\\opencti-platform\\opencti-graphql\\node_modules\\@elastic\\elasticsearch\\src\\api\\api\\index.ts:75:10)\n    at elIndex (C:\\Users\\LaurentBonnet\\IdeaProjects\\opencti\\opencti-platform\\opencti-graphql\\src\\database\\engine.js:2498:3)
\n    at createListTask (C:\\Users\\LaurentBonnet\\IdeaProjects\\opencti\\opencti-platform\\opencti-graphql\\src\\domain\\backgroundTask-common.js:184:3)"}],"inner_relation_creation":0,"level":"error","message":"Simple indexing fail","operation":"StixCoreObjectContainerTaskAddMutation","operation_query":"mutation StixCoreObjectContainerTaskAddMuta
tion($input:ListTaskAddInput!){listTaskAdd(input:$input){__typename id type}}","size":250,"time":18,"timestamp":"2024-01-05T10:03:02.586Z","type":"WRITE_ERROR","user":{"group_ids":["dbee0af4-7a7c-4722-ae3f-7e65202ccd2f"],"ip":"::1","organization_ids":[],"referer":"http://localhost:3000/dashboard/threats/threat_actors_group/9c393121-5c90-4049-9944-
095185ea7b6e","socket":"query","user_id":"88ec0c6a-13ce-5e39-b486-354fe4a7084f"},"variables":{"input":{"actions":[{"context":{"field":"container-object","options":{"includeNeighbours":true},"type":"ATTRIBUTE","values":["6df9caba-68b9-4840-abfc-2653e7fb8419"]},"type":"ADD"}],"ids":["9c393121-5c90-4049-9944-095185ea7b6e"],"scope":"KNOWLEDGE"}},"version":"5.12.15"}
```

image

**Comment by labo-flg** [comment]
Adding an intrusion set as Favorite using the "star" icon button in the Intrusion sets listing, fails with `update indexing failed`

```
"type": "strict_dynamic_mapping_exception",
"reason": "[1:803] mapping set to strict, dynamic introduction of [id] within [bookmarks] is not allowed"
```

log backend

```
{"category":"APP","errors":[{"attributes":{"body":{"script":{"params":{"bookmarks":[{"id":"be246fef-1f59-4d2c-808c-d54fd4e9ee9a","type":"Intrusion-Set"}],"entity_type":"User","internal_id":"88ec0c6a-13ce-5e39-b486-354fe4a7084f","updated_at":"2024-01-05T10:07:57.652Z"},"source":"ctx._source['internal_id'] = params['internal_id'];ctx._source['entity
_type'] = params['entity_type'];ctx._source['bookmarks'] = params['bookmarks'];ctx._source['updated_at'] = params['updated_at']"}},"documentId":"88ec0c6a-13ce-5e39-b486-354fe4a7084f","genre":"TECHNICAL","http_status":500},"message":"Update indexing fail","name":"DATABASE_ERROR","stack":"DATABASE_ERROR: Update indexing fail\n    at error (C:\\Users
\\LaurentBonnet\\IdeaProjects\\opencti\\opencti-platform\\opencti-graphql\\src\\config\\errors.js:8:10)\n    at DatabaseError (C:\\Users\\LaurentBonnet\\IdeaProjects\\opencti\\opencti-platform\\opencti-graphql\\src\\config\\errors.js:58:48)\n    at C:\\Users\\LaurentBonnet\\IdeaProjects\\opencti\\opencti-platform\\opencti-graphql\\src\\database\\e
ngine.js:2513:11\n    at processTicksAndRejections (node:internal/process/task_queues:95:5)\n    at async Promise.all (index 0)\n    at updateAttributeMetaResolved (C:\\Users\\LaurentBonnet\\IdeaProjects\\opencti\\opencti-platform\\opencti-graphql\\src\\database\\middleware.js:1938:7)\n    at addBookmark (C:\\Users\\LaurentBonnet\\IdeaProjects\\op
encti\\opencti-platform\\opencti-graphql\\src\\domain\\user.js:705:3)"},{"message":"strict_dynamic_mapping_exception\n\tRoot causes:\n\t\tstrict_dynamic_mapping_exception: [1:803] mapping set to strict, dynamic introduction of [id] within [bookmarks] is not allowed","name":"ResponseError","stack":"ResponseError: strict_dynamic_mapping_exception\n\
tRoot causes:\n\t\tstrict_dynamic_mapping_exception: [1:803] mapping set to strict, dynamic introduction of [id] within [bookmarks] is not allowed\n    at _SniffingTransport.request (C:\\Users\\LaurentBonnet\\IdeaProjects\\opencti\\opencti-platform\\opencti-graphql\\node_modules\\@elastic\\transport\\src\\Transport.ts:553:17)\n    at processTicksA
ndRejections (node:internal/process/task_queues:95:5)\n    at _Client.UpdateApi [as update] (C:\\Users\\LaurentBonnet\\IdeaProjects\\opencti\\opencti-platform\\opencti-graphql\\node_modules\\@elastic\\elasticsearch\\src\\api\\api\\update.ts:75:10)\n    at async Promise.all (index 0)\n    at updateAttributeMetaResolved (C:\\Users\\LaurentBonnet\\Id
eaProjects\\opencti\\opencti-platform\\opencti-graphql\\src\\database\\middleware.js:1938:7)\n    at addBookmark (C:\\Users\\LaurentBonnet\\IdeaProjects\\opencti\\opencti-platform\\opencti-graphql\\src\\domain\\user.js:705:3)"}],"inner_relation_creation":0,"level":"error","message":"Update indexing fail","operation":"StixDomainObjectBookmarkreateM
utation","operation_query":"mutation StixDomainObjectBookmarkreateMutation($id:ID!$type:String!){bookmarkAdd(id:$id type:$type){__typename id ...StixDomainObjectBookmark_node}}fragment StixDomainObjectBookmark_node on StixDomainObject{__isStixDomainObject:__typename id entity_type parent_types created_at updated_at modified ...on AttackPattern{nam
e x_mitre_id}...on Campaign{name}...on CourseOfAction{name}...on Individual{name}...on Organization{name}...on Sector{name}...on System{name}...on Indicator{name}...on Infrastructure{name}...on IntrusionSet{name countryFlag:stixCoreRelationships(relationship_type:\"originates-from\" toTypes:[\"Country\"]first:1 orderBy:created_at orderMode:desc){e
dges{node{to{__typename ...on Country{name x_opencti_aliases id}...on AdministrativeArea{id}...on Artifact{id}...on AttackPattern{id}...on AutonomousSystem{id}...on BankAccount{id}...on Campaign{id}...on CaseIncident{id}...on CaseRfi{id}...on CaseRft{id}...on CaseTemplate{id}...on Channel{id}...on City{id}...on CourseOfAction{id}...on Creator{id}.
..on CryptocurrencyWallet{id}...on CryptographicKey{id}...on CsvMapper{id}...on DataComponent{id}...on DataSource{id}...on Directory{id}...on DomainName{id}...on EmailAddr{id}...on EmailMessage{id}...on EmailMimePartType{id}...on EntitySetting{id}...on Event{id}...on ExternalReference{id}...on Feedback{id}...on Group{id}...on Grouping{id}...on Hos
tname{id}...on IPv4Addr{id}...on IPv6Addr{id}...on Incident{id}...on Indicator{id}...on Individual{id}...on Infrastructure{id}...on IntrusionSet{id}...on KillChainPhase{id}...on Label{id}...on Language{id}...on MacAddr{id}...on Malware{id}...on MalwareAnalysis{id}...on ManagerConfiguration{id}...on MarkingDefinition{id}...on MediaContent{id}...on 
Mutex{id}...on Narrative{id}...on NetworkTraffic{id}...on Note{id}...on ObservedData{id}...on Opinion{id}...on Organization{id}...on PaymentCard{id}...on PhoneNumber{id}...on Position{id}...on Process{id}...on Region{id}...on Report{id}...on Sector{id}...on Software{id}...on Status{id}...on StixCoreRelationship{id}...on StixFile{id}...on StixRefRe
lationship{id}...on StixSightingRelationship{id}...on System{id}...on Task{id}...on Text{id}...on ThreatActorGroup{id}...on ThreatActorIndividual{id}...on Tool{id}...on Url{id}...on UserAccount{id}...on UserAgent{id}...on Vulnerability{id}...on WindowsRegistryKey{id}...on WindowsRegistryValueType{id}...on Workspace{id}...on X509Certificate{id}}id}
}}avatar{id name}}...on Position{name}...on City{name}...on AdministrativeArea{name}...on Country{name}...on Region{name}...on Malware{name}...on ThreatActor{__isThreatActor:__typename name ...on ThreatActorIndividual{countryFlag:stixCoreRelationships(relationship_type:\"located-at\" toTypes:[\"Country\"]first:1 orderBy:created_at orderMode:desc){
edges{node{to{__typename ...on Country{name x_opencti_aliases id}...on AdministrativeArea{id}...on Artifact{id}...on AttackPattern{id}...on AutonomousSystem{id}...on BankAccount{id}...on Campaign{id}...on CaseIncident{id}...on CaseRfi{id}...on CaseRft{id}...on CaseTemplate{id}...on Channel{id}...on City{id}...on CourseOfAction{id}...on Creator{id}
...on CryptocurrencyWallet{id}...on CryptographicKey{id}...on CsvMapper{id}...on DataComponent{id}...on DataSource{id}...on Directory{id}...on DomainName{id}...on EmailAddr{id}...on EmailMessage{id}...on EmailMimePartType{id}...on EntitySetting{id}...on Event{id}...on ExternalReference{id}...on Feedback{id}...on Group{id}...on Grouping{id}...on Ho
stname{id}...on IPv4Addr{id}...on IPv6Addr{id}...on Incident{id}...on Indicator{id}...on Individual{id}...on Infrastructure{id}...on IntrusionSet{id}...on KillChainPhase{id}...on Label{id}...on Language{id}...on MacAddr{id}...on Malware{id}...on MalwareAnalysis{id}...on ManagerConfiguration{id}...on MarkingDefinition{id}...on MediaContent{id}...on
 Mutex{id}...on Narrative{id}...on NetworkTraffic{id}...on Note{id}...on ObservedData{id}...on Opinion{id}...on Organization{id}...on PaymentCard{id}...on PhoneNumber{id}...on Position{id}...on Process{id}...on Region{id}...on Report{id}...on Sector{id}...on Software{id}...on Status{id}...on StixCoreRelationship{id}...on StixFile{id}...on StixRefR
elationship{id}...on StixSightingRelationship{id}...on System{id}...on Task{id}...on Text{id}...on ThreatActorGroup{id}...on ThreatActorIndividual{id}...on Tool{id}...on Url{id}...on UserAccount{id}...on UserAgent{id}...on Vulnerability{id}...on WindowsRegistryKey{id}...on WindowsRegistryValueType{id}...on Workspace{id}...on X509Certificate{id}}id
}}}avatar{id name}}...on ThreatActorGroup{countryFlag:stixCoreRelationships(relationship_type:\"located-at\" toTypes:[\"Country\"]first:1 orderBy:created_at orderMode:desc){edges{node{to{__typename ...on Country{name x_opencti_aliases id}...on AdministrativeArea{id}...on Artifact{id}...on AttackPattern{id}...on AutonomousSystem{id}...on BankAccoun
t{id}...on Campaign{id}...on CaseIncident{id}...on CaseRfi{id}...on CaseRft{id}...on CaseTemplate{id}...on Channel{id}...on City{id}...on CourseOfAction{id}...on Creator{id}...on CryptocurrencyWallet{id}...on CryptographicKey{id}...on CsvMapper{id}...on DataComponent{id}...on DataSource{id}...on Directory{id}...on DomainName{id}...on EmailAddr{id}
...on EmailMessage{id}...on EmailMimePartType{id}...on EntitySetting{id}...on Event{id}...on ExternalReference{id}...on Feedback{id}...on Group{id}...on Grouping{id}...on Hostname{id}...on IPv4Addr{id}...on IPv6Addr{id}...on Incident{id}...on Indicator{id}...on Individual{id}...on Infrastructure{id}...on IntrusionSet{id}...on KillChainPhase{id}...
on Label{id}...on Language{id}...on MacAddr{id}...on Malware{id}...on MalwareAnalysis{id}...on ManagerConfiguration{id}...on MarkingDefinition{id}...on MediaContent{id}...on Mutex{id}...on Narrative{id}...on NetworkTraffic{id}...on Note{id}...on ObservedData{id}...on Opinion{id}...on Organization{id}...on PaymentCard{id}...on PhoneNumber{id}...on 
Position{id}...on Process{id}...on Region{id}...on Report{id}...on Sector{id}...on Software{id}...on Status{id}...on StixCoreRelationship{id}...on StixFile{id}...on StixRefRelationship{id}...on StixSightingRelationship{id}...on System{id}...on Task{id}...on Text{id}...on ThreatActorGroup{id}...on ThreatActorIndividual{id}...on Tool{id}...on Url{id
}...on UserAccount{id}...on UserAgent{id}...on Vulnerability{id}...on WindowsRegistryKey{id}...on WindowsRegistryValueType{id}...on Workspace{id}...on X509Certificate{id}}id}}}avatar{id name}}}...on Tool{name}...on Vulnerability{name}...on Incident{name}createdBy{__typename __isIdentity:__typename id name entity_type}objectLabel{edges{node{id valu
e color}}}objectMarking{edges{node{id definition_type definition x_opencti_order x_opencti_color}}}}","size":68,"time":91,"timestamp":"2024-01-05T10:07:57.686Z","type":"WRITE_ERROR","user":{"group_ids":["dbee0af4-7a7c-4722-ae3f-7e65202ccd2f"],"ip":"::1","organization_ids":[],"referer":"http://localhost:3000/dashboard/threats/intrusion_sets","socket":"query","user_id":"88ec0c6a-13ce-5e39-b486-354fe4a7084f"},"variables":{"id":"be246fef-1f59-4d2c-808c-d54fd4e9ee9a","type":"Intrusion-Set"},"version":"5.12.15"}

```

image

**Comment by labo-flg** [comment]
> Adding a Threat actor group to a container (a previously created incident response) - including neighbours, fails with Simple indexing failed

and

> Adding an intrusion set as Favorite using the "star" icon button in the Intrusion sets listing, fails with update indexing failed

Solved with [d053746](https://github.com/OpenCTI-Platform/opencti/pull/5340/commits/d05374678cbeed8dfe2ca7ea49d9aff4e6129524)

**Comment by labo-flg** [comment]
Failed to export data from Reports list (all reports - export-file-stix json exporter)

` mapping set to strict, dynamic introduction of [list_filters] within [metaData] is not allowed`

**Comment by labo-flg** [comment]
> Failed to export data from Reports list (all reports - export-file-stix json exporter)

Fixed.

**Comment by labo-flg** [comment]
I've fixed an issue on malware analysis with [e179687](https://github.com/OpenCTI-Platform/opencti/pull/5340/commits/e1796876656250e70fffde7324149aa21c621cd5) to be able to test their edition properly.
Same than what happened with log sources in Courses of Actions.

**Comment by labo-flg** [comment]
Cannot edit TAI's height or weight if there are none to begin with.

image

I solved this issue with [5eb9734](https://github.com/OpenCTI-Platform/opencti/pull/5340/commits/5eb9734f925d0a06d5269f68f2b34939008bec41)

**Pull request merged**